### DISCUSSION TITLE
Streaming API receiving large amount of zeros upon subscribing

### DISCUSSION THREAD

**Discussion by andig** [timdorr/tesla-api#359] [General]
I'm trying to reimplement the streaming api in go for sake of https://github.com/andig/evcc and always end up with

    stream read error: websocket: close 1009 (message too big)

Apparently Tesla sends back a large amount of zeros after the subscription message. I would appreciate any hint what might be causing this?

**Comment by ddaddy** [comment]
I had issues with the streaming socket disconnecting and it turned out to be an auth issue.
I was sending the bearer token from `https://auth.tesla.com/oauth2/v3/token` which is quite long (over 1000 char.) but I needed to be sending the token you can get from `https://owner-api.teslamotors.com/oauth/token` which is only approx 69 char. and starts with `qts-`.

**Comment by andig** [comment]
@ddaddy that‘s my mistake- I‘m using the bearer token, too. I could not find the oauth/token being used in the current code. Could you kindly point me to the relevant location?

**Comment by ddaddy** [comment]
Look at the `exchange_sso_access_token` function which exchanges the Bearer token for an owner-api token.

**Comment by andig** [comment]
Thank you, almost there:

    Sending: {"msg_type":"data:subscribe_oauth","token":"qts-...","value":"speed,odometer,soc,elevation,est_heading,est_lat,est_lng,power,shift_state,range,est_range,heading","tag":"191897433135"}
    stream received message: {"msg_type":"control:hello","connection_timeout":0}
    stream received message: {"msg_type":"data:error","tag":"191897433135","value":"owner_api error: `Got HTTP status: 403 Forbidden for . Response: {\"response\":\"unauthorized\"}`","error_type":"client_error"}

Not sure if the last is an error on my side or its due to being co-user of the car.

**Reply by ddaddy** [reply]
Looks like an incorrect tag. Use the `vehicle_id` and not the `id_s`

**Reply by andig** [reply]
That's it. Used the wrong vehicle id. Much appreciated!
